### PR TITLE
Add `WalletRead::get_seed_account`

### PR DESCRIPTION
**Pull request by nuttycom**
Builds atop #1258

### CODE REVIEW

File: zcash_client_sqlite/src/wallet.rs [hunk 1]
```diff
@@ -778,6 +778,46 @@ pub(crate) fn get_account_for_ufvk<P: consensus::Parameters>(
...
+    conn: &rusqlite::Connection,
+    params: &P,
+    seed: &HdSeedFingerprint,
+    account_id: zip32::AccountId,
+) -> Result<Option<(AccountId, Option<UnifiedFullViewingKey>)>, SqliteClientError> {
+    let mut stmt = conn.prepare(
+        "SELECT id, ufvk
+        FROM accounts
+        WHERE hd_seed_fingerprint = :hd_seed_fingerprint
+          AND hd_account_index = :account_id",
```
[the remark this answers is not in the document]
Comment (nuttycom) [reply zcash_client_sqlite/src/wallet.rs:794]: Fixed in https://github.com/zcash/librustzcash/pull/1259/commits/328e854202584be3055df4c9e1de915b48f4f1b5

File: zcash_client_sqlite/src/wallet/init/migrations/full_account_ids.rs [hunk 2]
```diff
@@ -69,6 +69,7 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
                     (account_type = {account_type_imported} AND hd_seed_fingerprint IS NULL AND hd_account_index IS NULL)
                 )
             );
+            CREATE UNIQUE INDEX hd_account ON accounts_new (hd_seed_fingerprint, hd_account_index);
```
Comment (str4d) [inline zcash_client_sqlite/src/wallet/init/migrations/full_account_ids.rs:72]: I don't understand the full subtleties of `UNIQUE INDEX` vs `CONSTRAIN UNIQUE` and why you'd choose one over the other. AFAICT:
- They both produce an index.
- `UNIQUE INDEX` can be removed and added without altering the table (and I guess `CREATE UNIQUE INDEX` fails if the table contains non-unique data?)
- `UNIQUE INDEX` can include expressions
- `CONSTRAIN UNIQUE` has greater support in `ON CONFLICT` logic (whereas `UNIQUE INDEX` only supports `INSERT ... ON CONFLICT` I think).

File: zcash_client_sqlite/src/wallet/init.rs [hunk 3]
```diff
@@ -436,6 +436,54 @@ mod tests {
             expected_idx += 1;
         }
 
+        let expected_indices = vec![
+            r#"CREATE UNIQUE INDEX accounts_ufvk ON "accounts" (ufvk)"#,
+            r#"CREATE UNIQUE INDEX accounts_uivk ON "accounts" (uivk)"#,
+            r#"CREATE UNIQUE INDEX hd_account ON "accounts" (hd_seed_fingerprint, hd_account_index)"#,
+            r#"CREATE INDEX "addresses_accounts" ON "addresses" (
```
Comment (str4d) [inline zcash_client_sqlite/src/wallet/init.rs:443]: Thanks for adding this; I was unaware that uniqueness constraints were starting to be embedded in the indices (which prior to now I only thought were present for performance). I rely heavily on changes to `init.rs` to track changes to the database structure and determine what constraints are applied.

**Review by str4d — APPROVED** [review]
utACK 79f5bb411ccc1baa4b0424b20ee66453ecb31016

File: zcash_client_sqlite/src/wallet/init/migrations/full_account_ids.rs (hunk 2 above)
Comment (nuttycom) [reply zcash_client_sqlite/src/wallet/init/migrations/full_account_ids.rs:72]: Yeah, I was looking at this today. I think that the `ON CONFLICT` capabilities are actually the same, from what I was reading, at least for recent releases? Given that the separate indices can be dropped and added independently, I think I have a slight preference for them, but I don't think it makes much difference one way or the other.

**Pull request merged**